tests/shared/src/test/scala/munit/AssertionsSuite.scala

```diff
@@ -50,12 +50,133 @@ class AssertionsSuite extends BaseSuite {
 
   test("subtype".tag(NoDotty)) {
     assertEquals(Option(1), Some(1))
+    assertEquals(Some(1), Option(1))
+    assertEquals(Option(1), Option(1))
+  }
+
+  test("false-negative") {
     assertNoDiff(
-      compileErrors("assertEquals(Some(1), Option(1))"),
-      """|error: Cannot prove that Option[Int] <:< Some[Int].
-         |assertEquals(Some(1), Option(1))
-         |            ^
-         |""".stripMargin
+      compileErrors("assertEquals(List(1), Vector(1))"),
+      if (isDotty)
+        """|error:
+           |Can't compare these two types:
+           |  First type:  List[Int]
+           |  Second type: Vector[Int]
+           |Possible ways to fix this error:
+           |  Alternative 1: provide an implicit instance for Compare[List[Int], Vector[Int]]
+           |  Alternative 2: upcast either type into `Any` or a shared supertype.
+           |I found:
+           |
+           |    munit.Compare.compareSubtypeWithSupertype[List[Int], Vector[Int]](
+           |      /* missing */summon[List[Int] <:< Vector[Int]]
+           |    )
+           |
+           |But no implicit values were found that match type List[Int] <:< Vector[Int].
+           |
+           |The following import might make progress towards fixing the problem:
+           |
+           |  import munit.CustomCompare.fromCustomEquality
```

Comment:
It's a shame these suggestions have false positives like this 🙈

Reply:
Mmmm, yeah this is a bummer.
I think it's related to https://github.com/lampepfl/dotty/issues/9685

Reply:
Nice find! I left a comment in that issue about the fact that we can reproduce without `inline` https://github.com/lampepfl/dotty/issues/9685#issuecomment-1100684424